src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/ResponseBuilder.kt

```diff
@@ -0,0 +1,65 @@
+package it.pagopa.ecommerce.helpdesk.utils
+
+import io.r2dbc.spi.Result
+import it.pagopa.generated.ecommerce.helpdesk.model.*
+import java.math.BigDecimal
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import org.reactivestreams.Publisher
+
+fun buildTransactionSearchResponse(
+    currentPage: Int,
+    totalCount: Int,
+    results: List<TransactionResultDto>
+): SearchTransactionResponseDto =
+    SearchTransactionResponseDto()
+        .page(PageInfoDto().current(currentPage).total(totalCount).results(results.size))
+        .transactions(results)
+
+fun resultToTransactionInfoDto(result: Result): Publisher<TransactionResultDto> =
+    result.map { row ->
+        TransactionResultDto()
+            .userInfo(
+                UserInfoDto()
+                    .userFiscalCode(row[0, String::class.java])
```

Comment:
If you want to make this a bit nicer you could define a `KtReadable` as follows:
```kotlin
data class KtReadable(val inner: Readable) {
    inline operator fun <reified T> get(index: Int): T? {
        return inner.get(index, T::class.java)
    }
}
```

wrap `row` with it

```kotlin
val row = KtReadable(row)
```

and then use it like

```suggestion
                    .userFiscalCode(row[0])
```

Reply:
Unfortunately implementations use reflection to convert row sql data to java objects. Kotlin types cannot be used here just because for some field a conversion in required (f.e. creationTimestamp and amounts).

I would let it as is, avoiding a mixed approach based on field type